modules/460-log-shipper/hooks/testdata/throttle-with-filter/result.json

```diff
@@ -0,0 +1,159 @@
+{
+  "sources": {
+    "cluster_logging_config/test-source": {
+      "type": "kubernetes_logs",
+      "extra_label_selector": "log-shipper.deckhouse.io/exclude notin (true)",
+      "extra_field_selector": "metadata.name!=$VECTOR_SELF_POD_NAME",
+      "extra_namespace_label_selector": "environment in (prod,test),log-shipper.deckhouse.io/exclude notin (true)",
+      "annotation_fields": {
+        "container_image": "image",
+        "container_name": "container",
+        "pod_ip": "pod_ip",
+        "pod_labels": "pod_labels",
+        "pod_name": "pod",
+        "pod_namespace": "namespace",
+        "pod_node_name": "node",
+        "pod_owner": "pod_owner"
+      },
+      "glob_minimum_cooldown_ms": 1000
+    }
+  },
+  "transforms": {
+    "transform/destination/test-es-dest/00_elastic_dedot": {
+      "drop_on_abort": false,
+      "inputs": [
+        "transform/source/test-source/01_clean_up"
+      ],
+      "source": "if exists(.pod_labels) {\n    .pod_labels = map_keys(object!(.pod_labels), recursive: true) -\u003e |key| { replace(key, \".\", \"_\") }\n}",
+      "type": "remap"
+    },
+    "transform/destination/test-es-dest/01_ratelimit": {
```

Comment:
There is a problem that each rate_limit transform has it's [own state](https://vector.dev/docs/reference/configuration/transforms/throttle/). Entries are limited separately. 

If you have a 500 entries limit and two rules in the exclude section with the following conditions for logs
1. Are not excluded.
2. Excluded only because of the first rate limit rule.
the actual limit of logs will be 1000.

To solve this, we need to find a way to build a single VRL for all excludes.

Reply:
ok, got it, I'll try to fix